src/main/java/org/tridiots/ipc/Server.java

```diff
@@ -1,60 +1,114 @@
         this.instance = instance;
-        this.serverSocket = new ServerSocket(port);
+        this.address = new InetSocketAddress("localhost", port);
+
+        this.accpetChannel = ServerSocketChannel.open();
+        this.accpetChannel.configureBlocking(false);
+        this.accpetChannel.socket().bind(address);
+
+        this.selector = Selector.open();
+        this.accpetChannel.register(selector, SelectionKey.OP_ACCEPT);
+
+        logger.info("Server start listen to server:" + address.getHostName() + ":" + address.getPort());
+    }
+
+    private void doAccept(SelectionKey key) throws IOException {
+        ServerSocketChannel server = (ServerSocketChannel) key.channel();
+        SocketChannel channel = server.accept();
+        channel.configureBlocking(false);
+
+        channel.register(selector, SelectionKey.OP_READ);
+        logger.debug("accept a connection from " + channel.socket().getInetAddress());
+    }
+
+    private void doRead(SelectionKey key) throws IOException, ClassNotFoundException {
+        SocketChannel channel = (SocketChannel) key.channel();
+
+        logger.debug("read from connection from " + channel.socket().getInetAddress().getHostName());
+        Param param = (Param)SocketObjectUtil.receiveObject(channel);
+
+        if (param == null) {
+            // the channel will always readable if we don't close
+            channel.close();
+            return;
+        }
+        Object result = call(param);
+        key.attach(result);
+
+        key.interestOps(SelectionKey.OP_WRITE);
     }
 
+    private void doWrite(SelectionKey key) throws IOException, ClassNotFoundException {
+        Object result =  key.attachment();
+        if (result == null) return;
+
+        SocketChannel channel = (SocketChannel) key.channel();
+        SocketObjectUtil.sendObject(channel, result);
+        key.interestOps(SelectionKey.OP_READ);
+
+        logger.debug("write to connection " + channel.socket().getInetAddress().getHostName());
+    }
+
+    private void doProcess(SelectionKey key) throws IOException {
+        ServerSocketChannel server = (ServerSocketChannel) key.channel();
+        SocketChannel channel = server.accept();
+
+        Param param = (Param)SocketObjectUtil.receiveObject(channel);
+        Object result = call(param);
+        SocketObjectUtil.sendObject(channel, result);
+        channel.shutdownOutput(); // otherwise the client will block on channel.read
+    }
     @Override
     public void run() {
         while (running) {
-            Socket socket = null;
             try {
-                socket = serverSocket.accept();
-                System.out.println("get connection from " + socket.getInetAddress().getHostName());
-            } catch (IOException e) {
-                e.printStackTrace();
-            }
+                if (selector.select() == 0) continue;
+                Iterator<SelectionKey> iter = this.selector.selectedKeys().iterator();
```

Comment:
As we have discussed before, we use `this` only when we assign something to fields.